File: KrishiSetu.Api/Models/Machinery.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;


namespace KrishiSetu.Api.Models
{
    public class Machinery
    {
        [Key]
        public int Id { get; set; }
        
        public int OwnerId { get; set; }
        public User Owner { get; set; } = null!;
        
        public int CategoryId { get; set; }
        public MachineryCategory Category { get; set; } = null!;
        
        [Required]
        public string Name { get; set; } = string.Empty;
        
        public string Description { get; set; } = string.Empty;
        
        public decimal RatePerHour { get; set; }
        public decimal RatePerDay { get; set; }
        
        public string AvailabilityStatus { get; set; } = "Available"; // Available, Booked, Maintenance
        
        public DateTime? AvailableDate { get; set; }
        
        public string? ImageUrl { get; set; }
        
        public bool IsApproved { get; set; } = false;

        public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    }
}

File: krishisetu-backend/src/main/java/com/krishisetu/backend/entity/Machinery.java
package com.krishisetu.backend.entity;

import jakarta.persistence.*;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "machinery")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Machinery {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private double ratePerHour;

    @Column(nullable = false)
    private double ratePerDay;

    private String imageUrl;

    private LocalDate availableDate;

    @JsonProperty("isApproved")
    private boolean approved = false;

    private boolean deleted = false;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id", nullable = false)
    private MachineryCategory category;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;
}
